### PR TITLE
Miscellaneous Fixes

### LINKED ISSUE #1622: Station alert console in engineering lobby is blank
Station alert console in engineering lobby, behind the desk, starts out empty. Needs to be deconstructed completely and put back together for it to have anything.

### LINKED ISSUE #1687: Lights destroyed by explosions cause runtimes in lighting visibility calculations
```
runtime error: 
[12:09:33]type mismatch: null -= /datum/light_source (/datum/light_source)
[12:09:33]proc name: smart vis update (/datum/light_source/proc/smart_vis_update)
[12:09:33]  source file: lighting_source.dm,309
[12:09:33]  usr: null
[12:09:33]  src: /datum/light_source (/datum/light_source)
[12:09:33]  call stack:
[12:09:33]/datum/light_source (/datum/light_source): smart vis update(1)
[12:09:33]/datum/light_source (/datum/light_source): update(space (221,110,1) (/turf/space))
[12:09:33]space (221,110,1) (/turf/space): update light()
[12:09:33]space (221,110,1) (/turf/space): set light(2, null, -99999, -99999, 0)
[12:09:33]space (221,110,1) (/turf/space): update starlight()
[12:09:33]the plating (220,109,1) (/turf/simulated/floor/plating): ChangeTurf(/turf/simulated/floor/plating (/turf/simulated/floor/plating), 1, 0)
[12:09:33]the plating (220,109,1) (/turf/simulated/floor/plating): ChangeTurf(/turf/simulated/floor/plating (/turf/simulated/floor/plating))
[12:09:33]the plating (220,109,1) (/turf/simulated/floor/plating): dismantle wall(null, null, null)
[12:09:33]the plating (220,109,1) (/turf/simulated/floor/plating): update damage()
[12:09:33]the plating (220,109,1) (/turf/simulated/floor/plating): take damage(234)
[12:09:33]the plating (220,109,1) (/turf/simulated/floor/plating): ex act(3)
[12:09:33]explosives (/datum/controller/process/explosives): explosion rec(space (222,111,1) (/turf/space), 18)
[12:09:33]explosives (/datum/controller/process/explosives): doWork()
[12:09:33]explosives (/datum/controller/process/explosives): process()
[12:09:33]/datum/controller/processSched... (/datum/controller/processScheduler): runProcess(explosives (/datum/controller/process/explosives))
```

### LINKED ISSUE #1824: Vampire thralls that are embraced aren't automatically given powers
As said, embraced thralls don't get their vampire tool-set automatically, even though they're classified as vampires.

Was embraced last night as a thrall, didn't get the vampire tab or verbs, Garn had to mess around with some shit and manually add verbs to me, one of which was on the lines of "check vampire powers" which gave me the base set of powers when used, after that everything worked normally.

Also, when brought to life as a former-thrall and now vampire, I wasn't frenzied either.

Given that there's even a specific thing when embracing about making a thrall your equal, this is likely a bug.

### LINKED ISSUE #1835: Shuttles are not playing nice with lighting overlays
Moving a shuttle causes numerous overlay nullspace pooling warnings, as well as causing strange behavior with BYOND-darkness inside the shuttle.
View-Variables reports the affected areas have lighting overlays; clearing+re-creating overlays does not fix, nor does triggering a lighting reconsideration.

Both long-jump shuttles such as the transfer shuttle and short-jump shuttles like the outpost shuttles have issues.

Observers are able to see into affected areas normally.

### LINKED ISSUE #1844: Drowsyness value can be negative
sleepytime

Probably related to sleeping toxin, also, it seems to leave said person forever stuck in said effects.

### PR DESCRIPTION
**Pull request by Lohikar**
changes:
- Engineering now actually gets engineering-type alert consoles instead of the non-functional base type.
- Lighting is no longer broken by shuttles.
- The ERT shuttle autoannouncer should no longer cause lighting runtimes.
- Fixed some bad newlines in some character setup help texts.
- Custom loadout tweaks now properly write to SQL.
- Custom loadout is now saved as JSON.
- Players are now shown a notice when the server was unable to load their custom loadout.
- Drowsyness is now clamped to zero.
- Added some new tips to the example.
- Tweaked some tips' grammar.
- Removed some tips that do not apply to our code base.
- Probably fixes embraced vampire thralls not getting vampire verbs.

Fixes #1844.
Fixes #1835.
Probably fixes #1687.
Probably fixes #1824.
Fixes #1622, #1521, and #1839. (Duplicates)

### CODE REVIEW

**Comment by Lohikar** [comment]
Death Commandos, Syndicate Commandos, and Monkeys are still technically in the game, but very uncommon. I don't really see the point of having tips for them, but I can re-add them if necessary.

As for "As an X", I deliberately broke away from the standard as a way of making the tips sound less similar to each other.

**Comment by LordFowl** [comment]
Monkeys are far more common than things such as borers. Half the point of tips is to shed knowledge on topics that are relatively unknown to players. Uncommon things should definitely be in there.

File: code/modules/client/preference_setup/loadout/loadout.dm [hunk 1]
```diff
@@ -66,9 +66,18 @@ var/list/gear_datums = list()
...
 /datum/category_item/player_setup_item/loadout/sanitize_character(var/sql_load = 0)
-
 	if (sql_load)
-		pref.gear	= params2list(pref.gear)
+		pref.gear_reset = FALSE
+		if (pref.gear && istext(pref.gear))
+			try
+				pref.gear = json_decode(pref.gear)
+			catch
+				world.log << "## SQL_CHAR: Unable to load preferences for client [pref.client ? pref.client.ckey : "UNKNOWN"]."
```
Comment (skull132) [inline code/modules/client/preference_setup/loadout/loadout.dm]: Debug message in log. :ree:

File: code/modules/client/preference_setup/loadout/loadout.dm [hunk 2]
```diff
@@ -106,6 +115,8 @@ var/list/gear_datums = list()
 		fcolor = "#E67300"
 	. = list()
 	. += "<table align = 'center' width = 100%>"
+	if (pref.gear_reset)
```
Comment (skull132) [inline code/modules/client/preference_setup/loadout/loadout.dm]: This `pref.gear_reset` should be declared within this object. Yes.

**Comment by Lohikar** [comment]
Requested changes made.

File: code/game/gamemodes/vampire/vampire_powers.dm [hunk 3]
```diff
@@ -850,6 +850,10 @@
 			if (alert(src, choice_text, "Choices", "Yes", "No") == "No")
 				src << "<span class='notice'>[denial_response]</span>"
 				return
+
+			vampire_thrall.remove_antagonist(T.mind, 0, 0)
+			qdel(draining_vamp)
```
Comment (skull132) [inline code/game/gamemodes/vampire/vampire_powers.dm]: Wait. Why are we deleting the vampire...?

**Pull request merged**